### PR TITLE
1531/feat/contact bank

### PR DESCRIPTION
**Pull request by y-golde**
todo
- [x] create new table script
- [x] connect contactperson fetching to new table
- [x] connect contactperson saving to new table
- [x] remove duplicated contact restrictions 
- [x] return contact bank
- [ ] see if you can re-arrange all contactedPersons somehow ( problem maybe the same person will appear several times) - see if you can flip the request so that person will be top level ---- OPEN BUG ON THE SUBJECT

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts [hunk 1]
```diff
@@ -178,37 +160,39 @@ const useContactQuestioning = (parameters: useContactQuestioningParameters): use
...
-                            occupation: contact.occupation,
-                            doesFeelGood: contact.doesFeelGood
-                                ? contact.doesFeelGood
+                            contactStatus: contact.personByPersonInfo.personContactDetailByPersonInfo?.contactStatus ?? NEW_CONTACT_STATUS_CODE,
+                            extraInfo: contact.personByPersonInfo.personContactDetailByPersonInfo?.extraInfo,
+                            relationship: contact.personByPersonInfo.personContactDetailByPersonInfo?.relationship,
+                            familyRelationship: contact.personByPersonInfo.personContactDetailByPersonInfo?.familyRelationship,
+                            isolationAddress: contact.personByPersonInfo.personContactDetailByPersonInfo?.isolationAddress,
+                            occupation: contact.personByPersonInfo.personContactDetailByPersonInfo?.occupation,
+                            doesFeelGood: contact.personByPersonInfo.personContactDetailByPersonInfo?.doesFeelGood
```
Comment (y-golde) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts:169]: change to server parsing

Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts:169]: changed :)

**Review by chenFriedman — APPROVED** [review]

File: server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts [hunk 2]
```diff
@@ -63,7 +63,7 @@ ContactedPeopleRoute.get('/allContacts/:minimalDateToFilter', handleInvestigatio
         .then(result => {
             allContactsLogger.info(validDBResponseLog, Severity.LOW);
             const allContactedPersons = result.data.allContactedPeople.nodes;
-            const convertedContacts = allContactedPersons.map((contact: InteractedContact) => ({...contact, ...contact.involvementReason}));
+            const convertedContacts = allContactedPersons.map((contact: any) => ({...contact, ...contact.involvementReason, ...contact.personByPersonInfo.personContactDetailByPersonInfo}));
```
Comment (barnahum8) [inline server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts:66]: why is the contact of type any?

Comment (y-golde) [reply server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts:66]: man this part is a mess to begin with: the truth is it's not really InteractedContact Either, this whole section needs to be re-typed, I'll open a bug on this subject and a seperate PR :)

Comment (barnahum8) [reply server/src/ClientToDBAPI/ContactedPeople/mainRoute.ts:66]: 👍

**Review by barnahum8 — APPROVED** [review]

**Pull request merged**